packages/jest-cli/src/TestNamePatternPrompt.js

```diff
@@ -30,108 +30,106 @@ const usage = () =>
 
 const usageRows = usage().split('\n').length;
 
-module.exports = (
-  config: Config,
-  pipe: stream$Writable | tty$WriteStream,
-  prompt: Prompt,
-) => {
-  class TestNamePatternPrompt {
-    // $FlowFixMe
-    _cachedTestResults;
-
-    constructor() {
-      (this: any).onChange = this.onChange.bind(this);
-    }
+module.exports = class TestNamePatternPrompt {
+  _cachedTestResults: Array<TestResult>;
+  _pipe: stream$Writable | tty$WriteStream;
+  _prompt: Prompt;
 
-    run(onSuccess: Function, onCancel: Function) {
-      pipe.write(ansiEscapes.cursorHide);
-      pipe.write(ansiEscapes.clearScreen);
-      pipe.write(usage());
-      pipe.write(ansiEscapes.cursorShow);
+  constructor(pipe: stream$Writable | tty$WriteStream, prompt: Prompt) {
+    this._pipe = pipe;
+    this._prompt = prompt;
+    (this: any).onChange = this.onChange.bind(this);
```

Comment:
In Flow you cannot overwrite class methods. I'll just get rid of this though, we don't need to bind this upfront really.